livekit-rtc/livekit/rtc/data_stream.py

```diff
@@ -0,0 +1,355 @@
+            size=header.total_length,
+            extensions=dict(header.extensions),
+            attachments=list(header.text_header.attached_stream_ids),
+        )
+        self._queue: RingQueue[proto_DataStream.Chunk | None] = RingQueue(capacity)
+        self._chunks: Dict[int, proto_DataStream.Chunk] = {}
+
+    def _on_chunk_update(self, chunk: proto_DataStream.Chunk):
+        self._queue.put(chunk)
+
+    def _on_stream_close(self, trailer: proto_DataStream.Trailer):
+        self._queue.put(None)
+
+    def __aiter__(self) -> AsyncIterator[TextStreamUpdate]:
+        return self
+
+    async def __anext__(self) -> TextStreamUpdate:
+        item = await self._queue.get()
+        if item is None:
+            raise StopAsyncIteration
+        decodedStr = item.content.decode()
+
+        self._chunks[item.chunk_index] = item
+        chunk_list = list(self._chunks.values())
+        chunk_list.sort(key=lambda chunk: chunk.chunk_index)
+        collected: str = "".join(map(lambda chunk: chunk.content.decode(), chunk_list))
+        return TextStreamUpdate(
+            current=decodedStr, index=item.chunk_index, collected=collected
+        )
+
+    @property
+    def info(self) -> TextStreamInfo:
+        return self._info
+
+    async def read_all(self) -> str:
+        final_string = ""
+        async for update in self:
+            final_string = update.collected
+        return final_string
+
+
+@dataclass
+class FileStreamInfo(BaseStreamInfo):
+    file_name: str
+    pass
+
+
+class FileStreamReader:
+    def __init__(self, header: proto_DataStream.Header, capacity: int = 0) -> None:
+        self._header = header
+        self._info = FileStreamInfo(
+            id=header.stream_id,
+            mime_type=header.mime_type,
+            topic=header.topic,
+            timestamp=header.timestamp,
+            size=header.total_length,
+            extensions=dict(header.extensions),
+            file_name=header.file_header.file_name,
+        )
+        self._queue: RingQueue[proto_DataStream.Chunk | None] = RingQueue(capacity)
+
+    def _on_chunk_update(self, chunk: proto_DataStream.Chunk):
+        self._queue.put(chunk)
+
+    def _on_stream_close(self, trailer: proto_DataStream.Trailer):
+        self._queue.put(None)
+
+    def __aiter__(self) -> AsyncIterator[bytes]:
+        return self
+
+    async def __anext__(self) -> bytes:
+        item = await self._queue.get()
+        if item is None:
+            raise StopAsyncIteration
+
+        return item.content
+
+    @property
+    def info(self) -> FileStreamInfo:
+        return self._info
+
+
+class BaseStreamWriter:
+    def __init__(
+        self,
+        local_participant: LocalParticipant,
+        topic: str = "",
+        extensions: Optional[Dict[str, str]] = {},
+        stream_id: str | None = None,
+        total_size: int | None = None,
+        mime_type: str = "",
+        destination_identities: List[str] = [],
```

Comment:
would like to make these kwargs only

```suggestion
        *,
        topic: str = "",
        extensions: Optional[Dict[str, str]] = {},
        stream_id: str | None = None,
        total_size: int | None = None,
        mime_type: str = "",
        destination_identities: List[str] = [],
```

Reply:
done!